src/server/index.ts

```diff
@@ -84,7 +84,9 @@ const serverOptions = (): ServerOptions => {
 }
 
 const registrationOptions = {
-  routes: { prefix: config.appPathPrefix }
+  routes: {
+    prefix: config.appPathPrefix === '' ? undefined : config.appPathPrefix
```

Comment:
Can probably let the `''` fall back to `undefined`?

Then we only have to remove the ` || undefined` in your improved PR

```suggestion
    prefix: config.appPathPrefix || undefined
```